libnavigation-core/src/test/java/com/mapbox/navigation/core/arrival/ArrivalProgressObserverTest.kt

```diff
@@ -341,9 +339,7 @@ internal class ArrivalProgressObserverTest {
                 every { arrivalInMeters } returns null
             }
         }
-        every { tripSession.updateLegIndex(any()) } returns mockk {
-            every { legIndex } returns 1
-        }
+        every { tripSession.updateLegIndex(any()) } returns true
```

Comment:
think it should be explicit here

```suggestion
        every { tripSession.updateLegIndex(1) } returns true
```

Reply:
because we are expecting invalid route leg updates to be false

``` kotlin
every { tripSession.updateLegIndex(0) } returns false
```